src/language_features/completion.rs

```diff
@@ -196,21 +192,23 @@ pub fn editor_completion(
             // we'll need to perform some transformations on the completion commands.
             if ctx.config.snippet_support && x.insert_text_format == Some(InsertTextFormat::SNIPPET)
             {
+                lazy_static::lazy_static! {
+                    static ref SNIPPET_TABSTOP_RE: Regex = Regex::new(r"\$(?P<i>\d+)").unwrap();
+                    static ref SNIPPET_PLACEHOLDER_RE: Regex = Regex::new(r"\$\{(?P<i>\d+):?(?P<placeholder>[^}]+)\}").unwrap();
+                }
                 let snippet = insert_text;
-                let insert_text = snippet_prefix_re
-                    .find(&snippet)
-                    .map(|x| x.as_str())
-                    .unwrap_or(&snippet);
+                let insert_text = SNIPPET_TABSTOP_RE.replace_all(&snippet, "");
+                let insert_text = SNIPPET_PLACEHOLDER_RE.replace_all(&insert_text, "$placeholder");
+                // Theres some issue with multiline insert texts, and they also dont work well in the UI, so display on one line
+                let insert_text = insert_text.replace('\n', "");
```

Comment:
sounds good. Do you happen to know an example of multiline insert texts? (not important, feel free to ignore)
I just observed that `if let` has no `insert_text` field, so the label is used, which is just one line.

Reply:
i don't. I did consider using the label as the actual completion text until the menu is closed, but the snippet text collapsed to a single line seemed like a better experience.